src/plugins/intel_cpu/src/transformations/snippets/x64/pass/snippets_mark_skipped.cpp

```diff
@@ -247,99 +248,83 @@ bool isSuitableChildForFusingSimple(const std::shared_ptr<const Node> &node, con
     // Note: Fusing child is allowed to have several users, but that must be the end of the chain
     return SupportsFusingWithConvolution_Simple(node, channelAxis) && getNumNonConstInputs(node) == 1;
 }
+
 bool isSuitableChildForFusingMatMul(const std::shared_ptr<const Node> &node, const bool canMatMulBeExecutedInI8,
                                     NodeFusingType &updatedChainType, int& fusingAxis) {
-    int num_non_const_inputs = 0;
-    bool can_be_converted_to_FC = false;
-    ov::PartialShape bias_shape;
-    ov::PartialShape matmul_shape;
-    for (const auto &parent_out : node->input_values()) {
-        const auto parent = parent_out.get_node_shared_ptr();
-        if (ov::op::util::is_on_constant_path(parent_out)) {
-            bias_shape = parent_out.get_shape();
-            num_non_const_inputs++;
-        } else {
-              matmul_shape = parent_out.get_partial_shape();
-              if (matmul_shape.size() == 0)
-                return false;
-            const auto& grandparents = parent->input_values();
-            // first check that weights are constant and both activations and weights have static shape
-            if (grandparents.size() == 2 &&
-                grandparents[1].get_partial_shape().is_static() &&
-                (ov::op::util::is_on_constant_path(grandparents[1]))) {
-                auto rank_a = grandparents[0].get_partial_shape().rank().get_length();
-                auto rank_w = grandparents[1].get_partial_shape().rank().get_length();
-                if (rank_a != 1 && rank_w != 1 && rank_a <= 3 && rank_w <= 3)
-                    can_be_converted_to_FC = true;
+    // Firsly check for Bias and DQScales fusion
+    const bool is_bias = ov::is_type<ov::opset1::Add>(node);
+    const bool is_dq_scales = ov::is_type<ov::opset1::Multiply>(node) && canMatMulBeExecutedInI8;
+    if (is_bias || is_dq_scales) {
+        for (const auto &in : node->inputs()) {
+            const auto& parent = in.get_source_output().get_node_shared_ptr();
+            const auto& parent_pshape = in.get_source_output().get_partial_shape();
+            if (ov::is_type<ov::op::v0::MatMul>(parent) && parent_pshape.rank().is_static()) {
+                if (parent->get_output_target_inputs(0).size() > 1)
+                    break;
+                const auto bias_port = 1 - in.get_index();
+                const auto bias_out = node->input_value(bias_port);
+                if (!ov::op::util::is_on_constant_path(bias_out) || (bias_out.get_target_inputs().size() > 1))
```

Comment:
Not sure that we will get some benefits from that in real scenarios, but can we perform less expensive checks first?   `ov::op::util::is_on_constant_path` can be really expensive if it is called on non constant subgraph, so I would prefer to see this function on the last place in each "or" condition

Reply:
Agree. I tried to do that in this PR :))) It's a root cause of the ticket.
But I'm not sure that it's good idea to that here because firstly we should know that `add` has constant input - and then make other checks. 
Moreover, to be honest, I haven't seen cases with `MatMuls` which has `Mul` or `Add` on outputs with inconstant second inputs. So I think it's safe to leave it so. What do you think?